### PR TITLE
Merging dev and prod branches

### PR DESCRIPTION
**Pull request by jatakiajanvi12**
- Changed the need to use an intermediate ACR for production in the azure build pipeline
- Created a new release deployment for merged branches

Scenarios tested: 
- Created a new azure build pipeline using .pipelines/azure_pipeline_mergedbranches.yaml. This works same as the dev pipeline.  Outputs the image to containerinsights.azurecr.io/public/azuremonitor/containerinsights/cidev. 
- Linux tag: <datetime>-<commitId> Windows tag: win-<datetime>-<commitId>
- Hooked this pipeline to a clone of the dev release pipeline that will put the image to AKS, ARC, Helm charts, etc. Images on the clusters also got updated as expected
- Also, created a new release pipeline and release to ciprod public MCR was successful.

### CODE REVIEW

**Review by pfrcks — COMMENTED** [review]
to be backwards compatible for the next 1-2 releases please make a new azure_pipeline yaml file.


this will ensure that if something goes wrong we can still use the the older way of doing things.  this should be followed for all new changes.

let's keep the option of using old workflow open. then after 1-2 sprints we can simply delete the older yamls, pipelines, etc

File: .pipelines/azure_pipeline_prod.yaml [hunk 1]
```diff
@@ -14,33 +14,32 @@ pr:
...
       commit=$(git rev-parse --short HEAD)
       datetime=$(date +'%m%d%Y')
-      linuxImagetag=ciprod-"$datetime"-"$commit"
-      windowsImageTag=win-ciprod-"$datetime"-"$commit"
+      linuxImagetag="$datetime"-"$commit"
+      windowsImageTag=win-"$datetime"-"$commit"
       echo "##vso[task.setvariable variable=linuxImagetag;isOutput=true]$linuxImagetag"
       echo "##vso[task.setvariable variable=windowsImageTag;isOutput=true]$windowsImageTag"
 
       cd $(Build.SourcesDirectory)/deployment/multiarch-agent-deployment/ServiceGroupRoot/Scripts
```
Comment (pfrcks) [inline .pipelines/azure_pipeline_prod.yaml:37]: is it needed for the new pipeline yaml?

Comment (jatakiajanvi12) [reply .pipelines/azure_pipeline_prod.yaml]: What do you mean? The tags will be used in docker build right?

**Comment by jatakiajanvi12** [comment]
> to be backwards compatible for the next 1-2 releases please make a new azure_pipeline yaml file.
> 
> this will ensure that if something goes wrong we can still use the the older way of doing things. this should be followed for all new changes.
> 
> let's keep the option of using old workflow open. then after 1-2 sprints we can simply delete the older yamls, pipelines, etc

Made this change.

File: .pipelines/azure_pipeline_prod.yaml (hunk 1 above)
Comment (pfrcks) [reply .pipelines/azure_pipeline_prod.yaml]: not sure why this comment is showing on this line. will place it at the right place

File: .pipelines/azure_pipeline_mergedbranches.yaml [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  steps:
+  - bash: |
+      commit=$(git rev-parse --short HEAD)
+      datetime=$(date +'%m%d%Y')
+      linuxImagetag="$datetime"-"$commit"
+      windowsImageTag=win-"$datetime"-"$commit"
+      echo "##vso[task.setvariable variable=linuxImagetag;isOutput=true]$linuxImagetag"
+      echo "##vso[task.setvariable variable=windowsImageTag;isOutput=true]$windowsImageTag"
+
+      cd $(Build.SourcesDirectory)/deployment/multiarch-agent-deployment/ServiceGroupRoot/Scripts
```
Comment (pfrcks) [inline .pipelines/azure_pipeline_mergedbranches.yaml:37]: we don't need this anymore correct?

Comment (pfrcks) [reply .pipelines/azure_pipeline_mergedbranches.yaml:37]: line 37 to be precise

Comment (jatakiajanvi12) [reply .pipelines/azure_pipeline_mergedbranches.yaml:37]: Yep right...Let me make this change

File: .pipelines/azure_pipeline_mergedbranches.yaml [hunk 3]
```diff
@@ -0,0 +1,285 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
```
Comment (ganga1980) [inline .pipelines/azure_pipeline_mergedbranches.yaml:2]: @jatakiajanvi12, did we test all the scenarios? if not, can we please test all the scenarios? if you have tested all the scenarios, can you please add to the PR description with list of scenarios being validated.

Comment (jatakiajanvi12) [reply .pipelines/azure_pipeline_mergedbranches.yaml:2]: @ganga1980 I have updated the description with a list. Let me know if you think I missed anything.

File: deployment/mergebranch-multiarch-agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+az login --identity
+if [ $? -eq 0 ]; then
+  echo "Logged in successfully"
+else
+  echo "-e error failed to login to az with managed identity credentials"
+  exit 1
+fi     
+
+echo "Pushing ${AGENT_IMAGE_FULL_PATH} to ${ACR_NAME}"
+az acr import --name $ACR_NAME --source mcr.microsoft.com/azuremonitor/containerinsights/cidev:${CDPX_TAG} --image $AGENT_IMAGE_FULL_PATH
```
Comment (pfrcks) [inline deployment/mergebranch-multiarch-agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:63]: nit: add as a variable

Comment (jatakiajanvi12) [reply deployment/mergebranch-multiarch-agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh]: @pfrcks Added the change

**Review by pfrcks — APPROVED** [review]
lgtm

**Pull request merged**